internal/daemon/controller/handlers/sessions/session_service.go

```diff
@@ -235,6 +235,8 @@ func (s Service) CancelSession(ctx context.Context, req *pbs.CancelSessionReques
 	if ses == nil {
 		return nil, handlers.NotFoundErrorf("Session %q doesn't exist.", req.GetId())
 	}
+	hostId := ses.HostId
+	hostSetId := ses.HostSetId
```

Comment:
I believe the context of the session host id and host set id are lost when canceling because the values are now stored in session_host_set_host. I will double check if there is something missing when calling cancel twice with the skip flag to will help us avoid setting these values manually.

Reply:
I think that I found an alternative solution, there is a function called `updateState` in the session repository that mutates the Session object when updating the row in the sessions table. This is when we lost the values for host_id and host_set_id. I can do a fetch those values in updateState.